### PR TITLE
docs: add cloud hosting docs

### PR DESCRIPTION
**Pull request by zalberico**
I wrote detailed cloud hosting documentation on my personal blog.

As part of an in progress grant I'm moving this documentation to the official site.

I've made some minor changes to better match the Urbit site's style.

Grant: https://grants.urbit.org/proposals/1452740478
Blog: https://zalberico.com/essay/2020/06/06/urbit-on-the-cloud.html

### CODE REVIEW

**Comment by zalberico** [comment]
@botter-nidnul for SA (I don't have permissions to add you as a reviewer in the sidebar).

I'm also having issues getting the zola static site generator to run so I haven't seen the generated html locally yet.

**Comment by matildepark** [comment]
@zalberico Check the [deploy preview](https://deploy-preview-466--urbit-org.netlify.app/) here.

**Comment by botter-nidnul** [comment]
The `-p 32123` argument needs to be added to the screen section. That's the only omission I see. 

More broadly though, I'm still not happy with nginx being used for this purpose and I'm not sure what the justification is. 

My concern is not just the extra steps it adds to the guide, but that it contradicts the Urbit philosophy (as I understand it). Nginx adds complexity to the running of a personal server, and increases management and maintenance difficulty. It's not just another piece of infrastructure Urbit is ~~parasitizing~~ epiphyting, nginx is sitting *above* Urbit, providing an essential function required for Urbit's basic operation (landscape), and duplicating a function that already exists in Urbit itself. 

Urbit hasn't been audited yet, but I don't think granting it the net_bind_service capability with setcap - so it can handle what nginx is doing here - is all that threatening on a simple, consumer run DO droplet. 

I think allowing Urbit access to port 80, and letting Urbit's acme handle the Let's Encrypt certificate requests, is best practice here.

**Review by baudtack — COMMENTED** [review]
The guide here is fine for what it is. But I share the concern about using nginx in a primary guide. I also don't think having hosting instructions for a specific host is something that should be hosted on urbit.org. Having these instructions is great, but they should be on someone's blog, not living on urbit.org as an endorsement of this company. Urbit should be neutral with regard to hosting. We can suggest possible hosts, but providing full instructions is too much.

**Comment by galenwp** [comment]
I think it'd be great to remove nginx as a part of this. But I disagree that something like this shouldn't be on urbit.org. It's ideal for people to host their planets in the cloud somewhere, I see no reason not to provide basic instructions on how to do so. If we had instructions for a few cloud hosts, that'd be great.

**Comment by matildepark** [comment]
> Having these instructions is great, but they should be on someone's blog, not living on urbit.org as an endorsement of this company.

I mean, Tlon.io represents the Tlon Corporation. Urbit.org is about the project; and hosting a ship is a basic documentation feature for the project specifically. Community docs and content will become more and more common ...

**Comment by matildepark** [comment]
Apparently Cmd+shift+enter closes an issue.

**Comment by zalberico** [comment]
Thanks for all of the feedback everyone :).

I'll refactor and remove the Nginx steps from the guide in favor of Urbit DNS arvo config (@botter-nidnul has helpfully written some of this up in a PR to my personal blog).

I'll also second the comment that it's important to have cloud hosting docs like this for new users. Getting a hosted urbit in the cloud is critical for new users being able to use urbit on multiple devices and a supported document we know works that makes things easier for them (and is kept up to date) is critical for removing that barrier.

I think over time this may become less true when there's an automated hosting option (Tlon is working on this) that is similar to what Matrix Element hosting is, but for now I think there's a lot of value in having explicit steps to guide the user and not forcing them to search the internet to find outdated or insecure setup instructions on their own.

[Edit: I think a note stating that urbit.org doesn't endorse any specific provider or company is also reasonable. My long term preference would be steps for the main providers, DO, AWS, Gcloud, Azure, maybe some home server raspberry pi option or something. I think getting urbit deployed on a server is a core part of the user experience and should be owned for it to be good]

**Comment by baudtack** [comment]
Is urbit.org going to maintain this document then? Does it have a droplet that this can be tested on? Is someone going to test this regularly to make sure these are still up to date?

**Comment by matildepark** [comment]
Urbit.org is being born, but I see it going that way, yeah. It's good to have a central point for editing as opposed to a million single-repo guides from years past.

**Comment by zalberico** [comment]
Ultimately yes, when it breaks we should fix it (and users will let us know).

**Comment by baudtack** [comment]
I just don't think it's the responsibility of urbit.org to teach people to use linux and we should be extremely aggressive in descoping things to avoid having a maintenance nightmare. This seems to be more a discussion of what the nature of urbit.org is and probably best for another place and with those working on it involved.

**Comment by zalberico** [comment]
I agree it's not urbit.org's responsibility to teach linux, but there is a responsibility to make urbit deployment easy for user adoption. 

I'm not sure a better path to get there other than explicit steps.

Ack on concern about scope expansion (definitely common problem/risk you're right to push back on), but I think one hosting example (even if it's only one) is helpful for making things way easier for users.

I'd argue not doing this actually incurs some future risk where centralized deployment services that abstract complexity away from the users entirely could gain ground.

[Edit: grammar]

**Comment by matildepark** [comment]
@jalehman opinions?

**Comment by jalehman** [comment]
> Urbit.org is about the project; and hosting a ship is a basic documentation feature for the project specifically. Community docs and content will become more and more common ...

I agree with this. Urbit.org should host a wide range of content that helps newcomers get set up, which means that it'll need to accommodate a wide range of different setups. A single end-to-end hosting example like this one has the potential to bring someone onto Urbit that wants to self-host, but is otherwise not proficient enough with Linux to get up and running.

> I just don't think it's the responsibility of urbit.org to teach people to use linux

It's definitely not our responsibility to teach anyone Linux, but as long as Linux is the only (edit: it's not the only one, but it is the *most* viable one) viable option to self-host your ship, the least we can do is, as a trusted authority, get someone past the Linux bits by telling them what to do to safely use Urbit in that environment.

> avoid having a maintenance nightmare

While having a maintenance nightmare is possible, it's not a foregone conclusion that one will occur if we expand the scope of the existing documentation. It's also possible that the number of contributors also expands and fills keeps this documentation updated as mistakes are encountered. If such a maintenance nightmare does occur, that's okay—I'm confident that we can find a path forward from there. We might then look to offload some of the more platform-specific bits to a platform that better supports user-generated content (e.g. a Wiki), rework the structure of urbit.org, or just delete documentation that is inaccurate and unmaintained.

Until we arrive at a maintenance nightmare though, I'd err on the side of admitting more documentation that stands to ease the onramp to Urbit.

**Comment by zalberico** [comment]
I've removed the nginx config. I also noticed that @botter-nidnul has put some work into integrating some DO docs already here: https://github.com/urbit/urbit.org/commit/0ec1521b303e3d75855cf48f5f80d6e8007d9a9b

@botter-nidnul - what do you think the right approach is? - it might make that install page simpler to break out the cloud hosting into its own section as I've done here and then just link to it from the install page. This page also goes into more of the user/domain/DNS setup which I think is useful to have in the core guide. Happy to merge these together or include anything from the work you've already done that you think I'm missing.

**Comment by zalberico** [comment]
I'm going to take a crack at simplifying the existing install docs (which have gotten a little unwieldy) as part of this PR.

Basic idea is to break out the hosting bits from the install page, will be clearer once I push up the changes.

**Comment by zalberico** [comment]
Apologies about the messy history, but I think things are finally in a good place to be reviewed.
I just reread it myself, but it'd be helpful to have another set of eyes go through it.

Main highlights:
- Link to the hosting document for cloud hosting, remove those steps from the install guide.
- Reduce a lot of repeated install instructions (makes the page more concise, reduces maintenance required for keeping docs up to date)
- Delete separate comet doc and put the handful of steps on install page (they were previously in both locations)
- Remove 'Make a Note' in favor of just having chat, once in chat users can figure out how to make a note and I don't think new users need that extra information in the install guide.
- Delete 'supernode' in favor just saying star or galaxy.
- Overall bias towards being concise as possible *while maintaining clarity*. 
- Fix typos

I think this is good to go, but feel free to comment on the changes.

**Comment by matildepark** [comment]
*pokes @baudtack *

File: content/using/install.md
Comment (jalehman) [inline content/using/install.md]: This is a good opportunity to link to the section about escaping your sponsor.

File: content/using/install.md
Comment (jalehman) [inline content/using/install.md]:
```suggestion
- **Keyfile**: Derived from the management proxy. Used as cryptographic proof that your Urbit ship is who it says it is. You use it to start up your ship for the first time.
```

**Comment by zalberico** [comment]
Once this merges, I'd also like to change the planet boot instructions so the ship is not inside the binary directory. This will make upgrading the binaries a lot simpler and less confusing because the pier will be separate.

Users can swap out the binary dir and it'll be clear that they can reuse their existing pier directory (I'll change the docs to do things this way by default). This should make binary upgrades less error prone for users.

I'd prefer to merge this PR first though since it's is already quite long and hard to review.

File: content/using/install.md
Comment (jalehman) [inline content/using/install.md]: Is a cloud service more convenient? It's more of a PITA in the short-term, but has all of the benefits listed above. I'm not sure this line is necessary.

File: content/using/install.md
Comment (jalehman) [inline content/using/install.md]: It seems like there's some redundancy here now.

File: content/using/install.md
Comment (jalehman) [inline content/using/install.md]:
```suggestion
Because of their need for near-perfect uptime Stars should be hosted on [cloud services](insert link here). 
```

**Review by jalehman — COMMENTED** [review]
A few initial thoughts.

File: content/using/install.md
Comment (zalberico) [reply content/using/install.md]: 👍 I was tempted to delete it, but I think it was already there in the mass of docs I removed. Will delete.

File: content/using/install.md
Comment (zalberico) [reply content/using/install.md]: Good catch - thanks

File: content/using/install.md
Comment (zalberico) [reply content/using/install.md]: On this one I didn't include the link because the next line has the link to the guide. I did switch it to your language though.

**Comment by zalberico** [comment]
Thanks for the review - I've committed all the changes.

File: content/using/install.md
Comment (baudtack) [inline content/using/install.md]: Object to saying comets are for temporary use.  Comets have many potential uses completely tangential to a sort of trial mode of urbit.

Reputation is not what prevents them from getting OTAs.

File: content/using/install.md
Comment (baudtack) [inline content/using/install.md]: Same objection to the phrasing here. Better might be "and can be used to try things out."

File: content/using/install.md
Comment (baudtack) [inline content/using/install.md:87]: extremely minor, but why was new removed?

File: content/using/install.md
Comment (baudtack) [inline content/using/install.md]: Should be noted here that this possibility of having to escape comes with buying a planet from _anyone_ not just opensea. You're more likely to have a problem there as it's not an urbit specific market, but it should be made clear this is not something specific to opensea.

File: content/using/install.md
Comment (baudtack) [inline content/using/install.md]: Why was paper removed?

File: content/using/install.md
Comment (baudtack) [inline content/using/install.md]: Let's not mention specific products for this.

Comment (baudtack) [reply content/using/install.md]: Honestly we should probably not for hardware wallets either.

**Review by baudtack — CHANGES_REQUESTED** [review]
Generally looks really good. Is there a pr for the docs repo side of this?

File: content/using/install.md
Comment (zalberico) [reply content/using/install.md]: What prevents them from getting OTAs? I asked on chat and someone told me it was because they don't get prioritized for compute since they have low default reputation.

Comment (zalberico) [reply content/using/install.md]: I'll change the language to remove 'temporary'

File: content/using/install.md
Comment (zalberico) [reply content/using/install.md:87]: I don't think it was - I think this is a github diff bug.

File: content/using/install.md
Comment (zalberico) [reply content/using/install.md]: I've read paper is just generally problematic (often destroyed/lost), I can add it back if you think we should have it.

File: content/using/install.md
Comment (zalberico) [reply content/using/install.md]: Will remove the specific companies

**Comment by zalberico** [comment]
I'll make these changes and push up a commit.

File: content/using/install.md
Comment (zalberico) [reply content/using/install.md]: I opted not to add paper back here (I thought of adding it with a flood/fire caveat - I've just read a lot of disaster stories of keys lost on paper), seems like if we're going to suggest things we might as well suggest the two better options.

**Comment by zalberico** [comment]
Thanks for the review feedback and comments.

I pushed up the changes.

**Comment by zalberico** [comment]
> Is there a pr for the docs repo side of this?

I think @matildepark took care of the changes on the docs repo side for this yesterday (I think the only needed change were the relative links in the glossary for comet).

I don't think there's anything else I need to do in the docs repo?

**Comment by matildepark** [comment]
https://github.com/urbit/docs/commit/ee4c4bbbf0ea19d89781bd73b7d7bff0408fd8e7 :P

**Comment by zalberico** [comment]
@baudtack if the changes look good to you, could you approve? Hoping to get this merged today. :)

**Review by baudtack — APPROVED** [review]
looks great!

**Pull request merged**
